services/search/utils.py

```diff
@@ -11,15 +11,35 @@ def create_synonym_list_esp(word_arr, max_synonyms=5):
             optional: Default value 5
     Returns: a list of synonyms
     """
+    session = FuturesSession()
+    reqs = create_conc_reqs(word_arr, session)
+
     synonyms = []
-    for word in word_arr:
-        payload = requests.get(constants.SPANISH_API_URL+word)
-        payload.raise_for_status()
 
-        word_synonyms = payload.json()['sinonimos']
+    for req in reqs:
+        resp = req.result()
+
+        word_synonyms = resp.json()['sinonimos']
+
         if len(word_synonyms) > max_synonyms:
             word_synonyms = word_synonyms[:max_synonyms]
         for synonym in word_synonyms:
             synonyms.append(synonym['sinonimo'])
 
     return synonyms
+
+
+def create_conc_reqs(word_arr, session):
```

Comment:
I think this would be a good time to add unit tests for the methods in the file. Looking at what's in `main`, it looks like this is untested right now.